File: jorgecunha07-frontend/src/app/dto/TaskDTO.ts
import {TaskStatusEnum} from "../Constants/TastStatusEnum";
import {TaskTypeEnum} from "../Constants/TaskTypeEnum";

export interface TaskDTO {
  id: number;
  taskStatus: TaskStatusEnum;
  description?: string;
  taskType: TaskTypeEnum;
  user: string;
  robotId: string;
  robotType: string;
  name: string;
}

File: jorgecunha07-mgt/DTO/TaskDto.cs
using MGT.Enums;

namespace MGT.DTO;

public class TaskDto
{
    public TaskStatusEnum TaskStatus { get; set; }
    public TaskTypeEnum TaskType { get; set; }
    public required string Description { get; set; }
    public required string User { get; set; }
    public string RobotId { get; set; }
    public string RobotType { get; set; }
    public required string Name { get; set; }
    public LocationDto FromLocation { get; set; }
    public LocationDto ToLocation { get; set; }
}
